File: 프로그래머스/lv2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []

    lst = []
    total = brown + yellow

    for w in range(3, total):
        if total % w == 0:
            lst.append([w, total // w])

    for exam in lst:
        if (exam[0] - 2) * (exam[1] - 2) == yellow:
            answer = exam

    return answer

File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;

vector<int> solution(int brown, int yellow) {
    vector<int> answer;
    
    ll boardSize = brown + yellow;

    for(int i = 3; i <= sqrt(boardSize); i++){
        if(boardSize % i == 0){
            int j = boardSize / i;
            
            if(brown == (j + i - 2) * 2){
                return {j, i};
            }
        }
    }
    
    return answer;
}